ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch_topology.go

```diff
@@ -75,6 +77,14 @@ func (topology ElasticsearchTopologyTF) payload(ctx context.Context, topologyID
 		return diags
 	}
 
+	if topology.InstanceConfigurationId.ValueString() != "" {
+		topologyElem.InstanceConfigurationID = topology.InstanceConfigurationId.ValueString()
+	}
+
+	if topology.InstanceConfigurationVersion.ValueInt64() > 0 {
```

Comment:
We can use `!topology.InstanceConfigurationVersion.IsUnknown()` instead to check whether plan has a value for the attribute. Also, I guess this is the answer to your question from the ticket.